### PR TITLE
Update 3ds2sdk release 2.3.1

### PR DESCRIPTION
**Pull request by robertdalmeida**
<changed> Updated 3ds2sdk version to 2.3.1  </changed>

### CODE REVIEW

File: AdyenActions/Components/3DS2/3DS2 SDK Adapters/AnyChallengeResult.swift [hunk 1]
```diff
@@ -11,7 +11,7 @@ internal protocol AnyChallengeResult {
 
     var sdkTransactionIdentifier: String { get }
 
-    var transactionStatus: String? { get }
+    var transactionStatus: String { get }
```
Comment (erenbesel) [inline AdyenActions/Components/3DS2/3DS2 SDK Adapters/AnyChallengeResult.swift:14]: This looks like a breaking change for v4, we may not be able to change this.

Also could be the reason why some jobs are failing.
Did you run the test target fully and did they all pass?

Comment (descorp) [reply AdyenActions/Components/3DS2/3DS2 SDK Adapters/AnyChallengeResult.swift:14]: It is internal 🤔

Comment (robertdalmeida) [reply AdyenActions/Components/3DS2/3DS2 SDK Adapters/AnyChallengeResult.swift:14]: The type `ADYChallengeResult` is public. Which means that it is a breaking change for checkout.
Discussed with @erenbesel. We can only include a version that doesn't have a breaking change. 

The max version without the breaking change is 2.2.6, which is the existing version.

Comment (descorp) [reply AdyenActions/Components/3DS2/3DS2 SDK Adapters/AnyChallengeResult.swift:14]: But nothing is breaking? Am I missing something?

Comment (robertdalmeida) [reply AdyenActions/Components/3DS2/3DS2 SDK Adapters/AnyChallengeResult.swift:14]: Now that you mentioned it. It is indeed not breaking for the consumer of checkout. 
But only for checkout who is a consumer of the 3ds2sdk. 
@erenbesel WDYT?

Comment (erenbesel) [reply AdyenActions/Components/3DS2/3DS2 SDK Adapters/AnyChallengeResult.swift:14]: Ohh my bad. Haven't noticed it's internal. All good :)
But jobs are failing, please look into them, I imagine some tests may fail due to this change.

Comment (descorp) [reply AdyenActions/Components/3DS2/3DS2 SDK Adapters/AnyChallengeResult.swift:14]: I have fixed Cocoapods. Will check Carthage..

**Comment by descorp** [comment]
Fixed up Carthage :)

**Review by descorp — APPROVED** [review]

**Review by erenbesel — APPROVED** [review]

**Review by neelSharma12 — APPROVED** [review]

**Pull request merged**